mgmtfn/dockplugin/dockplugin.go

```diff
@@ -469,6 +485,37 @@ func join() func(http.ResponseWriter, *http.Request) {
 func leave() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		logEvent("leave")
+
+		lr := api.LeaveRequest{}
+		content, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			httpError(w, "Could not read join request", err)
+			return
+		}
+
+		if err := json.Unmarshal(content, &lr); err != nil {
+			httpError(w, "Could not parse join request", err)
+			return
+		}
+
+		log.Infof("LeaveRequest: %+v", lr)
+
+		networkName, err := GetNetworkName(lr.NetworkID)
+		if err != nil {
+			log.Errorf("Error getting network name for UUID: %s. Err: %v", lr.NetworkID, err)
+			httpError(w, "Could not get network name", err)
+			return
+		}
+
+		// Delete the Endpoint
+		err = netPlugin.DeleteEndpoint(networkName + "-" + lr.EndpointID)
```

Comment:
I think we already delete the endpoint as part of delete which shall come after leave. Did anything change recently in libnetwork?